src/managers/CsvParser.java

```diff
@@ -0,0 +1,51 @@
+package managers;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvParser {
+
+    private final char separator;
+
+    public CsvParser(char separator) {
+        this.separator = separator;
+    }
+
+    public List<String[]> parse(String filePath) throws IOException {
+        List<String[]> records = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.trim().isEmpty()) {
+                    continue;
+                }
+                records.add(parseLine(line));
+            }
+        }
+        return records;
+    }
+
+    private String[] parseLine(String line) {
+        List<String> fields = new ArrayList<>();
+        StringBuilder currentField = new StringBuilder();
+        boolean inQuotes = false;
+
+        for (char ch : line.toCharArray()) {
+            if (ch == '\"') {
+                inQuotes = !inQuotes;
+            } else if (ch == separator && !inQuotes) {
+
+                fields.add(currentField.toString().trim());
+                currentField.setLength(0);
+            } else {
+                currentField.append(ch);
+            }
+        }
+
+        fields.add(currentField.toString().trim());
+        return fields.toArray(new String[0]);
```

Comment:
If you must use `new String[]` it is better to use 
```suggestion 
        return fields.toArray(new String[fields.size()]);
```